unit-tests/func/serializable-device/test-l500-json-load.py

```diff
@@ -50,48 +86,121 @@
-                continue
-            if expected_line != actual_line:
-                log.d( "Expected to find line:", expected_line)
-                log.d( "But found:            ", actual_line)
-    finally:
-        log.debug_unindent()
-
 test.start("Trying to load default settings from json")
 try:
-    sd.load_json(data)
+    sd.load_json( low_ambient_data_with_default_preset )
     visual_preset_number = depth_sensor.get_option(rs.option.visual_preset)
     visual_preset_name = rs.l500_visual_preset(int(visual_preset_number))
 
     # if this check fails it is most likely because FW changed the default settings
     equal = test.check_equal(visual_preset_name, rs.l500_visual_preset.low_ambient_light)
     if not equal:
         log.w( "It is possible that FW changed the default settings of the camera." )
-        log_settings_differences()
+        log_settings_differences( low_ambient_data_with_default_preset )
+except:
+    test.unexpected_exception()
+test.finish()
+
+#############################################################################################
+# this test checks that if the given settings are wrong (don't fit any preset), and we use default preset
+# we get custom preset
+
+wrong_data_with_default_preset = """
+{
+    "Alternate IR": 0.0,
+    "Apd Temperature": -9999,
+    "Confidence Threshold": 1,
+    "Depth Offset": 4.5,
+    "Depth Units": 0.000250000011874363,
+    "Digital Gain": 2,
+    "Enable IR Reflectivity": 0.0,
+    "Enable Max Usable Range": 0.0,
+    "Error Polling Enabled": 1,
+    "Frames Queue Size": 16,
+    "Freefall Detection Enabled": 1,
+    "Global Time Enabled": 0.0,
+    "Host Performance": 0.0,
+    "Humidity Temperature": 32.8908233642578,
+    "Inter Cam Sync Mode": 0.0,
+    "Invalidation Bypass": 0.0,
+    "LDD temperature": 32.1463623046875,
+    "Laser Power": 100,
+    "Ma Temperature": 39.667610168457,
+    "Mc Temperature": 31.6955661773682,
+    "Min Distance": 490,
+    "Noise Estimation": 0.0,
+    "Noise Filtering": 4,
+    "Post Processing Sharpening": 1,
+    "Pre Processing Sharpening": 0.0,
+    "Receiver Gain": 18,
+    "Reset Camera Accuracy Health": 0.0,
+    "Sensor Mode": 0.0,
+    "Trigger Camera Accuracy Health": 0.0,
+    "Visual Preset": 1
+}
+"""
+
+test.start("Trying to load wrong settings, should get custom preset")
+try:
+    sd.load_json( wrong_data_with_default_preset )
+    visual_preset_number = depth_sensor.get_option(rs.option.visual_preset)
+    visual_preset_name = rs.l500_visual_preset(int(visual_preset_number))
+
+    test.check_equal(visual_preset_name, rs.l500_visual_preset.custom)
+except:
+    test.unexpected_exception()
+test.finish()
+
+#############################################################################################
+# this test checks that if the given settings are wrong (don't fit any preset) but low-ambient-light preset is specified
```

Comment:
```
# Here, the json specifies a "low-ambient" preset -- so the values of the individual controls (that are part of this preset)
# should not matter. The result of loading the following should still result in LOW_AMBIENT even when the controls
# do not fit (because they're overriden).
```

Reply:
done